datastacks/utils.py

```diff
@@ -0,0 +1,94 @@
+import click
+import yaml
+
+from pydantic import BaseModel
+from datastacks.config_class import IngestConfig
+from jinja2 import Environment, FileSystemLoader
+from pathlib import Path
+
+
+def generate_target_dir(stage_name: str, dataset_name: str) -> str:
+    """Uses stage name and name of the dataset to generate the target directory to write to.
+
+    Args:
+        stage_name: Name of the pipeline stage eg. Ingest
+        dataset_name: Name of the dataset being processed
+
+    Returns:
+        Path to render template into
+    """
+    target_dir = f"de_workloads/{stage_name.lower()}/{stage_name.capitalize()}_{dataset_name.capitalize()}"
```

Comment:
Don't think we need to do the capitalize to the dataset name part (otherwise it changes `AzureSql_Generated_Example` to `Azuresql_generated_example`), happy for us to just leave that bit as it rather than doing some complex conversion.
```suggestion
    target_dir = f"de_workloads/{stage_name.lower()}/{stage_name.capitalize()}_{dataset_name}"
```